src/Fees.sol

```diff
@@ -71,17 +77,32 @@ abstract contract Fees is IFees, Owned {
     }
 
     /// @dev Only the lower 12 bits are used here to encode the fee denominator.
-    function _checkProtocolFee(uint16 fee) internal pure {
+    function _isFeeWithinBounds(uint16 fee) internal pure returns (bool) {
         if (fee != 0) {
             uint16 fee0 = fee % 64;
             uint16 fee1 = fee >> 6;
             // The fee is specified as a denominator so it cannot be LESS than the MIN_PROTOCOL_FEE_DENOMINATOR (unless it is 0).
             if (
                 (fee0 != 0 && fee0 < MIN_PROTOCOL_FEE_DENOMINATOR) || (fee1 != 0 && fee1 < MIN_PROTOCOL_FEE_DENOMINATOR)
             ) {
-                revert FeeTooLarge();
+                return false;
+            }
+        }
+        return true;
+    }
+
+    function _isValidProtocolFees(uint24 protocolFees) internal pure returns (bool) {
+        if (protocolFees != 0) {
+            uint16 protocolSwapFee = uint16(protocolFees >> 12);
+            uint16 protocolWithdrawFee = uint16(protocolFees & 0xFFF);
+            if (protocolSwapFee != 0 && !_isFeeWithinBounds(protocolSwapFee)) {
+                return false;
+            }
+            if (protocolWithdrawFee != 0 && !_isFeeWithinBounds(protocolWithdrawFee)) {
```

Comment:
it checks for non-0 here and then again inside `_isFeeWithinBounds`. I think it can just check once - i'd probably just keep the one in `_isFeeWithinBounds`

Reply:
actually i feel like you can just do 
```
return _isFeeWithinBounds(protocolSwapFee) && _isFeeWithinBounds(protocolWithdrawFee)
```

Reply:
Note that this will change when we have a PR for the changed fee model